ts/Core/Dynamics.ts

```diff
@@ -1485,6 +1516,8 @@ extend(Series.prototype, /** @lends Series.prototype */ {
                 (newType && newType !== this.type) ||
                 // New options affecting how the data points are built
                 typeof options.pointStart !== 'undefined' ||
+                chart.isInPlotOptions('pointInterval') ||
+                chart.isInPlotOptions('pointStart') ||
```

Comment:
It's not enough to check if option exists in `plotOptions`. We should keep points when value did not change. 

Also missing `pointIntervalUnit` and `keys` options too.